source/Playlist.cpp

```diff
@@ -55,9 +55,17 @@ void Playlist::Load(const DataNode &node)
 		else if(key == "location")
 			location.Load(child);
 		else if(key == "priority" && hasValue)
+		{
 			priority = max<unsigned>(0, child.Value(1));
+			if(child.Value(1) < 0)
+				child.PrintTrace("Warning: Negative weight sepcified for priority.");
+		}
 		else if(key == "weight" && hasValue)
+		{
 			weight = max<unsigned>(1, child.Value(1));
+			if(child.Value(1) < 0)
+				child.PrintTrace("Warning: Negative weight sepcified for priority.");
```

Comment:
```suggestion
				child.PrintTrace("Warning: Ignoring negative weight");
```
This shouldn't allow 0 either. It should allow fractional values, though, as that seems to be an arbitrary restriction.

Reply:
WeightedList uses only uint iirc

Reply:
Ah.